frame/democracy/src/lib.rs

```diff
@@ -525,33 +612,59 @@ decl_module! {
 
 		fn deposit_event() = default;
 
+		fn on_runtime_upgrade() -> Weight {
+			if StorageVersion::get() == Releases::V1 {
+				DepositOf::<T>::translate::<
+					(BalanceOf<T>, Vec<T::AccountId>), _
+				>(|_, (balance, accounts)| {
+					Some((accounts, balance))
+				});
+
+				StorageVersion::put(Releases::V2);
+
+				T::MaximumBlockWeight::get()
+			} else {
+				T::DbWeight::get().reads(1)
+			}
+		}
+
 		/// Propose a sensitive action to be taken.
 		///
 		/// The dispatch origin of this call must be _Signed_ and the sender must
 		/// have funds to cover the deposit.
 		///
 		/// - `proposal_hash`: The hash of the proposal preimage.
 		/// - `value`: The amount of deposit (must be at least `MinimumDeposit`).
+		/// - `proposals_upper_bound`: an upper bound on the current number of public proposals.
+		///   Extrinsic is weighted according to this value with no refund.
 		///
 		/// Emits `Proposed`.
 		///
 		/// # <weight>
-		/// - `O(P)`
-		/// - P is the number proposals in the `PublicProps` vec.
-		/// - Two DB changes, one DB entry.
+		/// - Complexity: `O(P)` where `P` is the number of proposals in the `PublicProps` vec.
+		/// - Db reads: `PublicPropCount`, `PublicProps`
```

Comment:
Where do we read `PublicProps`?

Reply:
PublicProps::append is counted as Db read + Db write.